src/assets/js/search.js

```diff
@@ -0,0 +1,164 @@
+(function($) {
+  'use strict';
+
+  /**
+   * Contains the search logic.
+   *
+   * @param searchLinks All elements which shall be included in the search.
+   * @constructor
+   */
+  function Search(searchLinks) {
+    this.searchLinks = searchLinks;
+  }
+
+  /**
+   * Searches the defined html elements and returns the result.
+   *
+   * Result format:
+   * {
+   *  "{group-name}": SearchResult[]
+   * }
+   *
+   * @param filter The search term
+   * @returns {{}}
+   */
+  Search.prototype.search = function(filter) {
```

Comment:
Please, use a more explicit function signature here, e.g. `Search.prototype.filter= function(pattern) {}` where `pattern` could be used with a RegExp to match elements.